packages/analytics-adapter/src/awaiter.ts

```diff
@@ -0,0 +1,80 @@
+
+export abstract class Awaiter {
+    public interval: number = 50;
+
+    constructor(public readonly maxWaitTime = 5000) {
+        /*
+        In the extending class, call the following in the constructor
+         this.waitForAnalytics();
+         this.waitForEvolv();
+        */
```

Comment:
Code smell: can't this superclass just call the methods on its own?

Reply:
@mattstrom -- would be good to discuss this one. I couldn't call them in the constructor for the superclass - as their implementations rely on things that are instantiated in the extending classes constructors. This means they run before the constructors in the extended classes